src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -44,1115 +44,716 @@
-                                        200.f / static_cast<float>(m_caps.wXmax - m_caps.wXmin);
-        state.axes[Joystick::Axis::Y] = (static_cast<float>(pos.dwYpos) -
-                                         static_cast<float>(m_caps.wYmax + m_caps.wYmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wYmax - m_caps.wYmin);
-        state.axes[Joystick::Axis::Z] = (static_cast<float>(pos.dwZpos) -
-                                         static_cast<float>(m_caps.wZmax + m_caps.wZmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wZmax - m_caps.wZmin);
-        state.axes[Joystick::Axis::R] = (static_cast<float>(pos.dwRpos) -
-                                         static_cast<float>(m_caps.wRmax + m_caps.wRmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wRmax - m_caps.wRmin);
-        state.axes[Joystick::Axis::U] = (static_cast<float>(pos.dwUpos) -
-                                         static_cast<float>(m_caps.wUmax + m_caps.wUmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wUmax - m_caps.wUmin);
-        state.axes[Joystick::Axis::V] = (static_cast<float>(pos.dwVpos) -
-                                         static_cast<float>(m_caps.wVmax + m_caps.wVmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wVmax - m_caps.wVmin);
-
-        // Special case for POV, it is given as an angle
-        if (pos.dwPOV != 0xFFFF)
-        {
-            const float angle                = static_cast<float>(pos.dwPOV) / 18000.f * 3.141592654f;
-            state.axes[Joystick::Axis::PovX] = std::sin(angle) * 100;
-            state.axes[Joystick::Axis::PovY] = std::cos(angle) * 100;
-        }
-        else
-        {
-            state.axes[Joystick::Axis::PovX] = 0;
-            state.axes[Joystick::Axis::PovY] = 0;
-        }
-
-        // Buttons
-        for (unsigned int i = 0; i < Joystick::ButtonCount; ++i)
-            state.buttons[i] = (pos.dwButtons & (1u << i)) != 0;
-    }
-
-    return state;
 }
 
 
 ////////////////////////////////////////////////////////////
-void JoystickImpl::initializeDInput()
+void JoystickImpl::DispatchRawInput(HRAWINPUT inputDevice)
 {
-    // Try to load dinput8.dll
-    dinput8dll = LoadLibraryA("dinput8.dll");
+    HRESULT hresult{};
+    UINT    bufferSize = sizeof(RAWINPUT);
+    UINT    uiResult{};
+    // a bigger stack-alloc, but since we spawned a separate thread for this we have plenty of stack memory
+    RAWINPUT  rawInput{};
+    HIDP_CAPS hCaps{};
+    DWORD     lastError = 0x0;
+
+    // It's nice being able to allocate an entire RAWINPUT on the stack, apparently RawInput has been around so long that folk used to malloc for it.
+    uiResult = GetRawInputData(inputDevice, RID_INPUT, &rawInput, &bufferSize, sizeof(RAWINPUTHEADER));
```

Comment:
```suggestion
    UINT uiResult = GetRawInputData(inputDevice, RID_INPUT, &rawInput, &bufferSize, sizeof(RAWINPUTHEADER));
```
It's best to declare a variable at the same place you initialize it rather than declaring it beforehand.